includes/class-amp-theme-support.php

```diff
@@ -1822,6 +1877,106 @@ public static function filter_customize_partial_render( $partial ) {
 		return $partial;
 	}
 
+	/**
+	 * Determine whether AMP dev mode is enabled.
+	 *
+	 * When enabled, the <html> element will get the data-ampdevmode attribute, and
+	 *
+	 * @since 1.3
+	 *
+	 * @return bool Whether AMP dev mode is enabled.
+	 */
+	public static function is_dev_mode_enabled() {
+
+		/**
+		 * Filters whether AMP mode is enabled.
+		 *
+		 * When enabled, the data-ampdevmode attribute will be added to the document element and it will allow the
+		 * attributes to be added to admin bar. It will also add the attribute to all elements which match the
+		 * queries for the expressions returned by the 'amp_dev_mode_element_xpaths' filter.
+		 *
+		 * @todo Should this be applied in add_hooks()? Ideally the admin-bar script would be skipped from being enqueued if dev mode is not going to be enabled.
+		 *
+		 * @since 1.3
+		 * @param bool[] Whether AMP dev mode is enabled.
+		 */
+		return apply_filters(
+			'amp_dev_mode_enabled',
+			(
+				// For the few sites that forcible show the admin bar even when the user is logged-out, only enable dev
+				// mode if the user is actually logged-in. This prevents the dev mode from being served to crawlers
+				// when they index the AMP version.
+				( is_admin_bar_showing() && is_user_logged_in() )
+				||
+				is_customize_preview()
+			)
+		);
+	}
+
+	/**
+	 * Get the XPath expressions to query for elements to add data-ampdevmode attributes.
+	 *
+	 * @since 1.3
+	 *
+	 * @return string[] XPath queries for elements to add data-ampdevmode attributes.
+	 */
+	private static function get_dev_mode_xpaths() {
+		/**
+		 * Filters the XPath queries for elements that should be enabled for dev mode.
+		 *
+		 * By supplying XPath queries to this filter, the data-ampdevmode attribute will automatically be added to the
+		 * root HTML element as well as to any elements that match the expressions. The attribute is added to the
+		 * elements prior to running any of the sanitizers.
+		 *
+		 * @since 1.3
+		 * @param string[] XPath element queries. Context is the root element.
+		 */
+		$dev_mode_xpaths = apply_filters( 'amp_dev_mode_element_xpaths', [] );
```

Comment:
I would cast to an array here, as you can never really trust plugins/themes to properly use filters.
```suggestion
		$dev_mode_xpaths = (array) apply_filters( 'amp_dev_mode_element_xpaths', [] );
```
This avoids the AMP plugin throwing errors in lines 1936-1937, and just make the offending plugin/theme not work as expected.